src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,162 @@
+﻿/*
+ * This class is adapted from IQKeyboardManager which is an open-source
+ * library implemented for iOS to handle Keyboard interactions with
+ * UITextFields/UITextViews. Link to their MIT License can be found here:
+ * https://github.com/hackiftekhar/IQKeyboardManager/blob/7399efb730eea084571b45a1a9b36a3a3c54c44f/LICENSE.md
+ */
+
+using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	// you can provide a topView argument or it will try to find the ContainerViewController
+	internal static void GoToNextResponderOrResign(UIView view, UIView? topView = null)
+	{
+		if (!view.CheckIfEligible())
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var targetSuperview = topView ?? view.GetViewController()?.GetContainerViewController();
+		if (targetSuperview is null)
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var textFields = GetDeepResponderViews(targetSuperview);
+
+		view.MoveToNextField(textFields);
+	}
+
+	static bool CheckIfEligible(this UIView view)
+	{
+		if (view is UITextField field && field.ReturnKeyType == UIReturnKeyType.Next)
+			return true;
+		else if (view is UITextView)
+			return true;
+
+		return false;
+	}
+
+	static UIView? GetContainerViewController(this UIViewController controller)
+	{
+		var curView = controller.View;
+
+		while (curView?.Superview is not null)
+		{
+			var curController = curView.GetViewController();
+			if (curController is ContainerViewController container)
+				return container.View;
+
+			curView = curView.Superview;
+		}
+
+		return null;
+	}
+
+	static void MoveToNextField (this UIView view, List<UIView> textFields)
+	{
+		// get the index of the current textField and go to the next one
+		var currentIndex = textFields.FindIndex(v => v == view);
+		var nextIndex = currentIndex < textFields.Count - 1 ? currentIndex + 1 : -1;
+
+		if (nextIndex != -1)
+			textFields[nextIndex].BecomeFirstResponder();
+		else
+			view.ResignFirstResponder();
+	}
+
+	// Find all of the eligible UITextFields and UITextViews inside this view
+	internal static List<UIView> GetDeepResponderViews(UIView view)
+	{
+		var textItems = view.GetResponderViews();
+
+		textItems.Sort(new ResponderSorter());
+
+		return textItems;
+	}
+
+	static List<UIView> GetResponderViews(this UIView view)
```

Comment:
What are your thoughts on making the search for the `Next` field part of the recursion here instead of collecting up all the fields? 

I feel like `GetDeepResponderViews` and `MoveToNextField` could be condensed. 

We know the location of the field we're starting at so when we call `GetDeepResponderViews` we can pick and choose through winners and losers as we search through the fields.  

That might also yield a general search method instead of something specific to this scenario. 

I waffle between 
- Good idea?
- Do we really need to over engineer this, plus that's not really possible because of....?